llama.cpp

```diff
@@ -5154,8 +5195,6 @@ struct llm_build_context {
 enum llm_offload_func_e {
     OFFLOAD_FUNC_NOP,
     OFFLOAD_FUNC,
-    OFFLOAD_FUNC_KQ,
-    OFFLOAD_FUNC_V,
```

Comment:
I cparams it looks like it is still possible to offload K and V separately, but wouldn't this change break that?

Reply:
Good point. 

Do we ever want to offload them separately? If not, I will change `cparams` to just `offload_kv`

Reply:
I am not sure, but looking at the difference in performance between 34 and 35 offloaded layers in the graphs in #3457 it is probably not worth supporting offloading V only.

Reply:
On the koboldcpp fork, loading just one of the two KV cache layers caused strange generation issues with seemingly random tokens appearing for a 13b.
I am not sure if this is a bug unique to the koboldcpp fork or if this is behavior similar to llama.cpp, but it helps support the argument that it's probably not worth supporting

Reply:
Ok, since both ways are equally easy to implement, I will not close the door for now and support separate K and V offloading, even if we don't see a use case at the moment

Reply:
I wonder if it makes a difference specifically in the context of models with/without GQA

Reply:
> Ok, since both ways are equally easy to implement,

It's actually simpler to go with wither both K and V offloaded or neither. Added `--no-kv-offload` cli arg